scan-results.js

```diff
@@ -0,0 +1,95 @@
+"use strict";
+
+const { URL } = require("url");
+
+const HIBP = require("./hibp");
+const sha1 = require("./sha1-utils");
+
+const scanResult = async(req, selfScan=false) => {
+
+  const allBreaches = req.app.locals.breaches;
+  let scannedEmail = null;
+
+  let foundBreaches = [];
+  let specificBreach = null;
+  let doorhangerScan = false;
+  let userCompromised = false;
+  let signedInUser = null;
+  let fullReport = false;
+  let userDash = false;
+
+
+  if (req.session.user) {
+    signedInUser = req.session.user;
+  }
+
+  // Checks if the user scanning their own verified email.
+  if (req.body && req.body.emailHash) {
+    scannedEmail = req.body.emailHash;
+    if (!selfScan && signedInUser) {
+      if (sha1(signedInUser.email).toUpperCase() === req.body.emailHash) {
+        selfScan = true;
```

Comment:
3 nested levels of `if` blocks may mean we should consider if we can simplify the logic here, or at least move it to its own helper function with a descriptive name?